controllers/nodemaintenance_controller.go

```diff
@@ -55,16 +72,379 @@ type NodeMaintenanceReconciler struct {
+
+	r.drainer = &drain.Helper{}
+
+	//Continue even if there are pods not managed by a ReplicationController, ReplicaSet, Job, DaemonSet or StatefulSet.
+	//This is required because VirtualMachineInstance pods are not owned by a ReplicaSet or DaemonSet controller.
+	//This means that the drain operation can’t guarantee that the pods being terminated on the target node will get
+	//re-scheduled replacements placed else where in the cluster after the pods are evicted.
+	//KubeVirt has its own controllers which manage the underlying VirtualMachineInstance pods.
+	//Each controller behaves differently to a VirtualMachineInstance being evicted.
+	r.drainer.Force = true
+
+	//Continue even if there are pods using emptyDir (local data that will be deleted when the node is drained).
+	//This is necessary for removing any pod that utilizes an emptyDir volume.
+	//The VirtualMachineInstance Pod does use emptryDir volumes,
+	//however the data in those volumes are ephemeral which means it is safe to delete after termination.
+	r.drainer.DeleteEmptyDirData = true
+
+	//Ignore DaemonSet-managed pods.
+	//This is required because every node running a VirtualMachineInstance will also be running our helper DaemonSet called virt-handler.
+	//This flag indicates that it is safe to proceed with the eviction and to just ignore DaemonSets.
+	r.drainer.IgnoreAllDaemonSets = true
+
+	//Period of time in seconds given to each pod to terminate gracefully. If negative, the default value specified in the pod will be used.
+	r.drainer.GracePeriodSeconds = -1
+
+	// TODO - add logical value or attach from the maintancene CR
+	//The length of time to wait before giving up, zero means infinite
+	r.drainer.Timeout = DrainerTimeout
+
+	// TODO - consider pod selectors (only for VMIs + others ?)
+	//Label selector to filter pods on the node
+	//r.drainer.PodSelector = "kubevirt.io=virt-launcher"
+
+	cs, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return err
+	}
+	r.drainer.Client = cs
+	r.drainer.DryRunStrategy = util.DryRunNone
+
+	r.drainer.Out = writer{klog.Info}
+	r.drainer.ErrOut = writer{klog.Error}
+	r.drainer.OnPodDeletedOrEvicted = onPodDeletedOrEvicted
+	return nil
+}
+
+func (r *NodeMaintenanceReconciler) checkLeaseSupported() error {
+	isLeaseSupported, err := checkLeaseSupportedInternal(r.drainer.Client)
+	if err != nil {
+		r.logger.Error(err, "Failed to check for lease support")
+		return err
+	}
+	r.isLeaseSupported = isLeaseSupported
+	return nil
+}
+
+func (r *NodeMaintenanceReconciler) setOwnerRefToNode(instance *nodemaintenancev1beta1.NodeMaintenance, node *corev1.Node) {
+
+	for _, ref := range instance.ObjectMeta.GetOwnerReferences() {
+		if ref.APIVersion == node.TypeMeta.APIVersion && ref.Kind == node.TypeMeta.Kind && ref.Name == node.ObjectMeta.GetName() && ref.UID == node.ObjectMeta.GetUID() {
+			return
+		}
+	}
+
+	// logger is nil... r.logger.Info("setting owner ref to node")
```

Comment:
need to resole this

Reply:
but of course not holding this pr. maybe just leave a todo. your call

Reply:
forgot about this one, thx for the heads up. Seems the logger was only missing during unit tests, so I just added it